mailpoet/tasks/GithubClient.php

```diff
@@ -58,13 +59,25 @@ public function downloadReleaseZip($zip, $downloadDir, $tag = null) {
     if (!is_dir($downloadDir)) {
       mkdir($downloadDir, 0777, true);
     }
-    $this->httpClient->get($assetDownloadUrl, ['sink' => $downloadDir . $zip, 'headers' => ['Accept' => 'application/octet-stream']]);
+    $this->get($assetDownloadUrl, ['sink' => $downloadDir . $zip, 'headers' => ['Accept' => 'application/octet-stream']]);
     file_put_contents($downloadDir . '/' . $zip . '-info', $assetDownloadInfo);
   }
 
   private function getRelease($tag = null) {
     $path = 'releases/' . ($tag && $tag !== 'latest' ? "tags/$tag" : 'latest');
-    $response = $this->httpClient->get($path);
+    $response = $this->get($path);
     return json_decode($response->getBody()->getContents(), true);
   }
+
+  private function get($uri, array $options = []): ResponseInterface {
+    $retries = 0;
+    do {
+      $response = $this->httpClient->get($uri, $options);
+      if ($response->getStatusCode() < 500) {
+        return $response;
+      }
```

Comment:
I see that [the Guzzle's documentation mentions](https://docs.guzzlephp.org/en/latest/quickstart.html#exceptions) that for 500 there is an GuzzleHttp\Exception\ServerException thrown. I'm not super familiar with Guzzle, maybe this behaviour is configurable. I just wanted to double-check with you if it is ok without trying/catching.

Reply:
Thank you. I was trying to find this in the documentation but failed; I will fix this.